aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/MilestoneStepTest.java

```diff
@@ -0,0 +1,166 @@
+package org.jenkinsci.plugins.workflow.steps;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.support.steps.MilestoneStepExecution;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import static org.junit.Assert.*;
+
+import hudson.model.Result;
+
+public class MilestoneStepTest {
+
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    @Rule
+    public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    @Before
+    public void clear() {
+        MilestoneStepExecution.clear();
```

Comment:
:ant: do not copy my bad example; store state in an `Extension`-scoped instance field.